System/PathFinder.cpp

```diff
@@ -318,31 +334,40 @@ namespace RTE {
-		if (node->Down) { node->DownCost = CostAlongLine(node->Pos + Vector(-3, 0), node->Down->Pos + Vector(-3, 0)); }
-		if (node->Left) { node->LeftCost = std::max(node->Left->RightCost, CostAlongLine(node->Pos + Vector(0, -3), node->Left->Pos + Vector(0, -3))); }
+		if (node->Up) { node->UpMaterial = getStrongerMaterial(node->Up->DownMaterial, StrongestMaterialAlongLine(node->Pos + Vector(3, 0), node->Up->Pos + Vector(3, 0))); }
+		if (node->Right) { node->RightMaterial = StrongestMaterialAlongLine(node->Pos + Vector(0, 3), node->Right->Pos + Vector(0, 3)); }
+		if (node->Down) { node->DownMaterial = StrongestMaterialAlongLine(node->Pos + Vector(-3, 0), node->Down->Pos + Vector(-3, 0)); }
+		if (node->Left) { node->LeftMaterial = getStrongerMaterial(node->Left->RightMaterial, StrongestMaterialAlongLine(node->Pos + Vector(0, -3), node->Left->Pos + Vector(0, -3))); }
 
-		if (node->UpRight) { node->UpRightCost = std::max(node->UpRight->DownLeftCost, CostAlongLine(node->Pos + Vector(2, 2), node->UpRight->Pos + Vector(2, 2))); }
-		if (node->RightDown) { node->RightDownCost = CostAlongLine(node->Pos + Vector(2, -2), node->RightDown->Pos + Vector(2, -2)); }
-		if (node->DownLeft) { node->DownLeftCost = CostAlongLine(node->Pos + Vector(-2, -2), node->DownLeft->Pos + Vector(-2, -2)); }
-		if (node->LeftUp) { node->LeftUpCost = std::max(node->LeftUp->RightDownCost, CostAlongLine(node->Pos + Vector(-2, 2), node->LeftUp->Pos + Vector(-2, 2))); }
+		if (node->UpRight) { node->UpRightMaterial = getStrongerMaterial(node->UpRight->DownLeftMaterial, StrongestMaterialAlongLine(node->Pos + Vector(2, 2), node->UpRight->Pos + Vector(2, 2))); }
+		if (node->RightDown) { node->RightDownMaterial = StrongestMaterialAlongLine(node->Pos + Vector(2, -2), node->RightDown->Pos + Vector(2, -2)); }
+		if (node->DownLeft) { node->DownLeftMaterial = StrongestMaterialAlongLine(node->Pos + Vector(-2, -2), node->DownLeft->Pos + Vector(-2, -2)); }
+		if (node->LeftUp) { node->LeftUpMaterial = getStrongerMaterial(node->LeftUp->RightDownMaterial, StrongestMaterialAlongLine(node->Pos + Vector(-2, 2), node->LeftUp->Pos + Vector(-2, 2))); }
 
 		// Mark this as already changed so the above expensive calculation isn't done redundantly
 		node->IsUpdated = true;
 
 		for (int i = 0; i < PathNode::c_MaxAdjacentNodeCount; ++i) {
-			float delta = std::abs(oldCosts[i] - node->AdjacentNodeCosts[i]);
-			if (delta > c_NodeCostChangeEpsilon) {
+			const Material *oldMat = oldMaterials[i];
+			const Material *newMat = node->AdjacentNodeBlockingMaterials[i];
+
+			// Check if the material strength is more than our delta, or if a door has appeared/dissappeared (since we handle their costs in a special manner)
+			float delta = std::abs(oldMat->GetIntegrity() - newMat->GetIntegrity());
+			bool doorChanged = oldMat != newMat || (oldMat->GetIndex() == g_MaterialDoor || newMat->GetIndex() == g_MaterialDoor);
```

Comment:
Why does the doorChanged bool true if the old material isn't the new material? This can happen for stuff other than doors changing, right?
Maybe there's a better name for this bool?

Or is that 1st || supposed to be an &&?